### ISSUE TITLE
Crash when tapping marker after orientation change

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#3517]
Instructions to reproduce:
 - Open the `Press for marker`-Activity in the test app
 - Long press to add a marker
 - Click marker to open infowindow
 - Rotate device (infowindow is closed. marker is visible)
 - Click marker results in crash

>
java.lang.NullPointerException: Attempt to invoke virtual method 'android.graphics.PointF com.mapbox.mapboxsdk.views.NativeMapView.pixelForLatLng(com.mapbox.mapboxsdk.geometry.LatLng)' on a null object reference
                                                                                at com.mapbox.mapboxsdk.views.MapView.toScreenLocation(MapView.java:2101)
                                                                                at com.mapbox.mapboxsdk.annotations.InfoWindow.open(InfoWindow.java:99)
                                                                                at com.mapbox.mapboxsdk.annotations.Marker.showInfoWindow(Marker.java:108)
                                                                                at com.mapbox.mapboxsdk.annotations.Marker.showInfoWindow(Marker.java:104)
                                                                                at com.mapbox.mapboxsdk.views.MapView.selectMarker(MapView.java:2518)
                                                                                at ...

That trace points to:
```java
     PointF point = mNativeMapView.pixelForLatLng(location);
```
Inside this method:
```java
    @UiThread
    @NonNull
    public PointF toScreenLocation(@NonNull LatLng location) {
        if (location == null) {
            Log.w(TAG, "location was null, so just returning (0, 0)");
            return new PointF();
        }

        PointF point = mNativeMapView.pixelForLatLng(location);

        float x = point.x * mScreenDensity;
        float y = point.y * mScreenDensity;

        // flip y direction vertically to match core GL
        y = getHeight() - y;

        return new PointF(x, y);
    }
```

**Comment by tobrun** [comment]
This is a very strange issue, this only occurs on markers that have opened the InfoWindow once.
Others work as expected. 

During debugging I noticed that `mNativeMapView` is null in those cases but I could also verify that mNativeMapView was succesfully created in `onCreate` before and `onDestroy` was not called.

**Comment by tobrun** [comment]
This crash only occurs in `PressForMarkerActivity`:

That activity is able to restore Markers, added by the user, after an activity recreation.
```java
if (savedInstanceState != null) {
     mMarkerList = savedInstanceState.getParcelableArrayList(STATE_MARKER_LIST);
     mMapView.addMarkers(mMarkerList);
}
```
Probably a problem with serialisation or recreation.

**Comment by tobrun** [comment]
The issue was that an InfoWindow was not correctly recreated with the new MapView

**Comment by tobrun** [comment]
This landed with 2631f5b in release branch of `3.1.0` #3537
